### PR TITLE
Remove threshold from from getBalances method

### LINKED ISSUE #212: getBalances will no longer require a threshold
### Bug description
In this [PR](https://github.com/iotaledger/iri/pull/1768#pullrequestreview-365391016), the `threshold` parameter is removed from the getBalances endpoint on IRI nodes.

Currently, this is a required parameter is the library, so we should make it optional.

### Expected behaviour
The `threshold` parameter is optional (for backwards compatibility)

### Actual behavior
The `threshold` parameter is required

### Am I planning to do it myself with a PR?
No.

### PR DESCRIPTION
**Pull request by ezienecker**
# Description of change

Remove threshold from `getBalances` method 
fixes #212

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Adjust test they uses the deprecated method.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests using ginkgo that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kwek20** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Iota-java/iota-java/pullRequest?prid=5560218&bid=18235007)

**Comment by ezienecker** [comment]
Are there any other todos that need to be made?

**Comment by kwek20** [comment]
What do you mean @ezienecker ? 
For this PR, were just waiting for 1.8.6 release of IRI (RCs have already been released)

**Comment by ezienecker** [comment]
OK, I thought I was something missing, any remarks.

**Comment by kwek20** [comment]
Ill give it a more thorough review on Monday,as this isnt urgent :=)
But a cursory scan showed good stuff, i liked the change  from `nodes != null && nodes.size() > 0` to `!nodes.isEmpty()`

File: jota/src/main/java/org/iota/jota/IotaAPICore.java [hunk 1]
```diff
@@ -76,11 +108,9 @@ public boolean addNode(Connection n) {
                         return true;
                     }
                 }
-                
+
                 boolean started = n.start();
                 if (started) {
-                
-                    //Huray! Lets add it
```
Comment (kwek20) [inline jota/src/main/java/org/iota/jota/IotaAPICore.java:83]: :(

**Comment by kwek20** [comment]
Two small things; can you remove the `threshold` in `IotaNodeHTTPService:157`, and Deprecate Constants.INVALID_THRESHOLD_ERROR @ezienecker ? Thank you!

**Comment by ezienecker** [comment]
@kwek20 I'll take a look at it today

File: jota/src/main/java/org/iota/jota/IotaAPICore.java (hunk 1 above)
Comment (ezienecker) [reply jota/src/main/java/org/iota/jota/IotaAPICore.java:83]: Do we need this?

**Comment by ezienecker** [comment]
@kwek20 I implemented your suggestions.

File: jota/src/main/java/org/iota/jota/IotaAPICore.java (hunk 1 above)
Comment (kwek20) [reply jota/src/main/java/org/iota/jota/IotaAPICore.java:83]: No :)

**Pull request merged**